### ISSUE TITLE
Processing fails if preprocess = FALSE, or no products are specified

### ISSUE THREAD

**Issue by lbusett** [ranghetti/sen2r#259]
Crashes following: 

``` r
out1 <- sen2r(
    gui     = FALSE,
    extent  = system.file("extdata/vector/scalve.kml", package = "sen2r"),
    timewindow = c("2019-04-29", "2019-04-29"),
    preprocess = FALSE,
    path_l1c = safe_folder,
    path_l2a = safe_folder,
    path_out = out_folder_1, 
    s2tiles_selected = "32TNR"
)
```

with: 

```
[2020-01-22 11:52:38] Starting sen2r execution.
[2020-01-22 11:52:38] Searching for available SAFE products on SciHub...
[2020-01-22 11:52:41] Starting to download the required level-2A SAFE products.
Images S2B_MSIL2A_20190429T101029_N0211_R022_T32TNR_20190429T132900.SAFE are already on your system and will be skipped. Set `overwrite_safe` to TRUE to re-download them
No L2A images needed
[2020-01-22 11:52:41] Download of level-2A SAFE products terminated.
[2020-01-22 11:52:41] Starting to download the required level-1C SAFE products.
Images S2B_MSIL1C_20190429T101029_N0207_R022_T32TNR_20190429T125333.SAFE are already on your system and will be skipped. Set `overwrite_safe` to TRUE to re-download them.
No L1C images needed
[2020-01-22 11:52:41] Download of level-1C SAFE products terminated.
Error in { : 
  task 1 failed - "task 1 failed - "object 'safe_names_l1c_tocorrect' not found""
```
The same happens even if we set `s2_levels = "l1c"` 

If no products are specified, images are not downloaded.: 

``` r
out1 <- sen2r(
    gui     = FALSE,
    extent  = system.file("extdata/vector/scalve.kml", package = "sen2r"),
    timewindow = c("2019-04-29", "2019-04-29"),
    path_l1c = safe_folder,
    path_l2a = safe_folder,
    path_out = out_folder_1, 
    s2tiles_selected = "32TNR"
)
```
```
> `[2020-01-22 11:57:55] Starting sen2r execution.
> [2020-01-22 11:57:55] Searching for available SAFE products on SciHub...
> [2020-01-22 11:57:55] No SAFE products matching the settings were found.
>  
> [2020-01-22 11:57:56] ###### sen2r Processing Report ###### 
> [2020-01-22 11:57:56] Dates to be processed based on processing parameters: 0
> [2020-01-22 11:57:56] Processing completed for: all expected dates.
> [2020-01-22 11:57:56] ###### Execution of sen2r session terminated ######
> The processing chain can be re-launched with the command:
>   sen2r("/home/lb/.sen2r/proc_par/s2proc_20200122_115755.json")
> Warning messages:
> 1: Parameter "s2_levels" was not specified; setting it to the default ('l1c').Parameter "s2_levels" was not specified; setting it to the default ('l2a'). 
> 2: In set(x, j = name, value = value) :
>   Column 'footprint' does not exist to remove`
```

I guess this is expected behaviour, but we should probably improve reporting in this case (That is: if preprocess = TRUE, but no expected products, just issue a warning/stop and return immediately)

**Comment by lbusett** [comment]
Concerning the last part: 

> I guess this is expected behaviour, but we should probably improve reporting in this case (That is: if preprocess = TRUE, but no expected products, just issue a warning/stop and return immediately)

I already implemented something in https://github.com/ranghetti/sen2r/commit/5a1980c790ed021ac338f65d18c4eacda4d37f57

**Comment by ranghetti** [comment]
Thanks, fixed with d60e140eebcd9ae802aea0168af9b6df99657597 and 10097e69974188f9441fa4c36ce03acde00813b9.
